server/src/main/java/org/elasticsearch/transport/TransportStats.java

```diff
@@ -99,15 +96,15 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeVLong(rxSize);
         out.writeVLong(txCount);
         out.writeVLong(txSize);
-        if (out.getTransportVersion().onOrAfter(TransportVersions.V_8_1_0)) {
-            assert (inboundHandlingTimeBucketFrequencies.length > 0) == (outboundHandlingTimeBucketFrequencies.length > 0);
+        assert (inboundHandlingTimeBucketFrequencies.length > 0) == (outboundHandlingTimeBucketFrequencies.length > 0);
+        if (out.getTransportVersion().before(TransportVersions.TRANSPORT_STATS_HANDLING_TIME_REQUIRED)) {
             out.writeBoolean(inboundHandlingTimeBucketFrequencies.length > 0);
```

Comment:
This doesn't match up with the reading code which requires these things to have length `HandlingTimeTracker.BUCKET_COUNT`.

Reply:
I've hardened up the assertions on the bucket frequencies length in efc71fd0334535a74845b31adec0c20a7773b316